code-challenges.js

```diff
@@ -16,41 +22,137 @@ const fruit2 = "banana"
+console.log(boilingPoint(temp3))// returned "212 at boiling point"
+
+// I ended up getting a missing Initializer declaration for boilingPoint because I didn't put an = after boilingPoint to give it meaning with the function
 
 // --------------------2) Create the code that will combine the two arrays and return the length using the variables provided below. Expected output: 10
 
 // Pseudo code:
+// Start by concating the 2 arrays together wrapped in a console.log
+// Next, create a new const variable for the new array that was concacted of myNumbers1 and myNumbers2.
+// Then, console.log new const to check the output.
+// For final step, console.log new array with the .push setter for arrays.
 
 const myNumbers1 = [3, 7, 0, 36, -9]
 const myNumbers2 = [8, -7, 42, 9, 13]
 
+console.log(myNumbers1.concat(myNumbers2))// output [3, 7, 0, 36, -9, 8, -7, 42, 9, 13
+const myNumbers3 = myNumbers1.concat(myNumbers2) 
+console.log(myNumbers3) // output [3, 7, 0, 36, -9, 8, -7, 42, 9, 13]
+console.log(myNumbers3.length) // output 10
 
 // --------------------3) Create the code that will reverse the letters of a string using the test variable provided below. Expected output: "2202 atleD"
 
 // Pseudo code:
-
+// Create a new a variable set equal to the variable name .reverse
+    // Console.log new variable
+    // The output produced an error stating .reverse is not a function
+    //Back to the pseudo coding drawing board.
+// Attempt #2 in the form of a function expression.
+    // Declare a function with the parameter of split to pass currentCohort through.
+    // Use the return with the parameter of string.split("") to turn the currentCohort string into an array of letters.
+    // Create a variable equal to the output of function
+    // Console.log the new variable to get the output
+    // Create a variable equal to the previous variable .reverse
+    // Console.log this new variable with the .reverse
+    // Then, I was going to use the .join method to put it all back together, but this isn't working.
+    // const split = (string) => {
+    //     return string.split("")
+    //  }
+    // console.log(split(currentCohort)) // Output ["D", "e", "l", "t", "a", " ", "2", "0", "2", "2"]
+    // // Now we have an array of elements instead of a string
+    
+    // var array = console.log(split(currentCohort))
+    // console.log(array) // // Output ["D", "e", "l", "t", "a", " ", "2", "0", "2", "2"]
+    // var reverseArray = array.reverse()
+    // console.log(reverseArray) // TypeError: Cannot read properties of undefined (reading 'reversed')
+//Attempt #3
+// Research new approaches to the problem
+// Between the syllabus notes and some other reseaarch, I am finding I need to write a function using the keyword function. Can you explain more about this syntax and the difference between a function expression and function declaration? Is the keyword function being used to define an expression here? Just trying to grasp the theory.
```

Comment:
This can be a little tricky to explain but the two are very similar.  You can write your example below with function notation as well and it should work.  I do find this article to be pretty helpful in showing some of the main differences between the two with some examples: [Function declarations vs expressions](https://medium.com/@mandeep1012/function-declarations-vs-function-expressions-b43646042052)